src/test/ui/methods/receiver-ptr-mutability.rs

```diff
@@ -0,0 +1,20 @@
+// run-pass
+
+#![feature(arbitrary_self_types)]
+
+struct Test {}
+
+impl Test {
+    fn x(self: *const Self) {
+        println!("x called");
+    }
+}
+
+fn main() {
+    let ptr: *mut Test = std::ptr::null_mut();
+    ptr.x();
```

Comment:
Hmm, we should probably a test that checks that the generated MIR doesn't have any auto-ref style coercions. @oli-obk what's the recommended way to do that sort of thing these days...?

Reply:
You can put the test into `src/tests/mir-opt`. There are instructions for dumping the MIR of functions in the README in the same folder. I think you'll want to dump the `mir_built` MIR (right after MIR building, without any transformations having been run).

Reply:
Done. Could you review the test please? Relevant MIR in the test output:
Done. Could you review the test please? Relevant MIR in the test output:

```
// First call
        _4 = _1;
        _3 = move _4 as *const Test (Pointer(MutToConstPointer));
        StorageDead(_4);
        _2 = Test::x(move _3) -> [return: bb2, unwind: bb4];

// Second call
        _12 = (*(*(*(*_5))));
        _11 = move _12 as *const Test (Pointer(MutToConstPointer));
        StorageDead(_12);
        _10 = Test::x(move _11) -> [return: bb3, unwind: bb4];
```

These have right number of derefs and ascriptions.